spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -1469,12 +1483,20 @@ public void testPartitionsTableLastUpdatedSnapshot() {
         new GenericRecordBuilder(
             AvroSchemaUtil.convert(
                 partitionsTable.schema().findType("partition").asStructType(), "partition"));
+
+    List<DataFile> dataFilesFromFirstCommit = listDataFilesFromCommitId(table, firstCommitId);
```

Comment:
Would it work to make a method List<DataFile> dataFiles(table) to get all the data files, so we don't have to do add data files from both commits?

I did this before here: https://github.com/apache/iceberg/blob/master/spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java#L682

(I think we can do it without column stats here).

If we do this, we can even extract to TestHelpers in a later PR.

Reply:
@szehon-ho Didn't know this approach, thanks!

> If we do this, we can even extract to TestHelpers in a later PR.
+1, let's do the extraction in a later PR.